google/cloud/aiplatform/training_jobs.py

```diff
@@ -616,181 +908,233 @@ def run(
+                Higher values in this column give more importance to the row
+                during Model training. The column must have numeric values between 0 and
+                10000 inclusively, and 0 value means that the row is ignored.
+                If the weight column field is not set, then all rows are assumed to have
+                equal weight of 1.
+            budget_milli_node_hours (int):
+                Optional. The train budget of creating this Model, expressed in milli node
+                hours i.e. 1,000 value in this field means 1 node hour.
+                The training cost of the model will not exceed this budget. The final
+                cost will be attempted to be close to the budget, though may end up
+                being (even) noticeably smaller - at the backend's discretion. This
+                especially may happen when further model training ceases to provide
+                any improvements.
+                If the budget is set to a value known to be insufficient to train a
+                Model for the given training set, the training won't be attempted and
+                will error.
+                The minimum value is 1000 and the maximum is 72000.
+            model_display_name (str):
+                Optional. If the script produces a managed AI Platform Model. The display name of
+                the Model. The name can be up to 128 characters long and can be consist
+                of any UTF-8 characters.
+            disable_early_stopping (bool):
+                Required. If true, the entire budget is used. This disables the early stopping
+                feature. By default, the early stopping feature is enabled, which means
+                that training might stop before the entire training budget has been
+                used, if futrher training does no longer brings significant improvement
+                to the model.
 
         Returns:
-            model: AI Platform Model if training succeeded and produced an AI Platform
-                Model. None otherwise.
+            model: The trained AI Platform Model resource or None if training did not
+                produce an AI Platform Model.
 
         Raises:
-            RuntimeError if Training failed.
+            RuntimeError if Training job has already been run
         """
-        self._block_until_complete()
 
-        if self.has_failed:
-            raise RuntimeError(
-                f"Training Pipeline {self.resource_name} failed. No model available."
-            )
+        training_task_definition = schema.training_job.definition.tabular_task
+
+        training_task_inputs_dict = {
+            # required inputs
+            "targetColumn": target_column,
+            "transformations": self._column_transformations,
+            "trainBudgetMilliNodeHours": budget_milli_node_hours,
+            # optional inputs
+            "weightColumnName": weight_column,
```

Comment:
In [google3/google/cloud/aiplatform/publicfiles/trainingjob/definition/automl_tables.proto](https://source.corp.google.com/piper///depot/google3/google/cloud/aiplatform/publicfiles/trainingjob/definition/automl_tables.proto;rcl=342552818;l=21), this column is referred to as "weight_column_name".

However, in `gs://google-cloud-aiplatform/schema/trainingjob/definition/automl_tabular_1.0.0.yaml`, it gives the name as "weightColumn". This is incorrect and results in an error.

@sasha-gitg why is there a discrepancy here?

Reply:
Following up on this.

Reply:
Synced with the team. The yaml is incorrect and will be updated. The field should be ```weightColumnName```. If users are using our Model Builder SDK then hopefully we have caught these issues early during our development. This should become less of an issue for the service as we move forward because it will support all previously versioned yaml schemas.